app/helpers/spotlight/main_app_helpers.rb

```diff
@@ -27,4 +27,18 @@ def field_enabled? field, *args
     field.enabled && field.send((:show if controller.is_a? Blacklight::Catalog and ["edit", "show"].include?(action_name)) || document_index_view_type)
   end
 
+  def link_back_to_catalog(opts={:label=>nil})
```

Comment:
I assume you ran into these problems after fixing the permissive visibility one?